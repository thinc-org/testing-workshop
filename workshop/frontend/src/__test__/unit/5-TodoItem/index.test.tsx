import { fireEvent, render } from "@testing-library/react"

describe("TodoItem", () => {
  const mockText = "text"
  const onChangeSpy = jest.fn()

  const mockProps = {
    text: mockText,
    checked: false,
    onChange: onChangeSpy,
  }

  const handleOpenSpy = jest.fn()
  const useDialogSpy = jest.fn(() => ({
    handleOpen: handleOpenSpy,
  }))

  jest.doMock("../4-useDialog", () => useDialogSpy)

  afterEach(() => {
    jest.clearAllMocks()
  })

  it("should render correctly", async () => {
    const { default: TodoItem } = await import(".")

    const { container } = render(<TodoItem {...mockProps} />)
    expect(container).toMatchSnapshot()
  })

  it("should render black text if checked is false", async () => {
    const { default: TodoItem } = await import(".")

    const { getByText } = render(<TodoItem {...mockProps} />)

    const element = getByText(mockText)
    expect(element).toHaveStyle({ color: "#999" })
  })

  it("should render gray text if checked is true", async () => {
    const { default: TodoItem } = await import(".")

    const { getByText } = render(<TodoItem {...mockProps} checked={true} />)

    const element = getByText(mockText)
    expect(element).toHaveStyle({ color: "#000" })
  })

  it("should call handleOpen if the button was clicked", async () => {
    const { default: TodoItem } = await import(".")

    const { container } = render(<TodoItem {...mockProps} />)

    const button = container.querySelector("button")
    expect(button).toBeDefined()

    if (button) fireEvent.click(button)
    expect(handleOpenSpy).toBeCalledTimes(1)
  })
})
