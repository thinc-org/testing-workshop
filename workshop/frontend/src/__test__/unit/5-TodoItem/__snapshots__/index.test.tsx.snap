// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render correctly 1`] = `
<div>
  <div>
    <input
      type="checkbox"
    />
    <p
      style="color: rgb(153, 153, 153);"
    >
      text
    </p>
    <button>
      delete
    </button>
  </div>
</div>
`;
